.github/workflows/check-headers.yml

```
@@ -0,0 +1,32 @@
+name: Headers
+
+on:
+  pull_request:
+
+permissions: {}
+
+jobs:
+  check-headers:
+    name: Check headers
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          # we don't know what commit the last tag was it's safer to get entire repo so previousStableVersion resolves
+          fetch-depth: 0
+
+      - name: Setup Java 8
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 8
+
+      - name: Cache Coursier cache
+        uses: coursier/cache-action@v6.4.0
+
+      - name: Enable jvm-opts
+        run: cp .jvmopts-ghactions .jvmopts
+
+      - name: Check headers
+        run: sbt +headerCheckAll
```

Probably also needs `http-tests / MultiJvm / headerCheck`.

---

It doesn't aggregate so the project needs to be specified manually:

```suggestion
        run: sbt +headerCheckAll "+;http-tests/MultiJvm/headerCheck"
```

---

So I just tried locally and `sbt +headerCheckAll` also seems to work with multi jvm tests? To do this I manually removed the header from `http-tests/src/multi-jvm/scala/org/apache/pekko/http/PekkoHttpServerLatencyMultiNodeSpec.scala`, ran `sbt +headerCheckAll` and it failed, i.e.

```
<@incubator-pekko-http>-<⎇ create-header-check-workflow-and-remove-from-validatePullRequest>-<*>-> sbt +headerCheckAll
[info] welcome to sbt 1.8.2 (Amazon.com Inc. Java 1.8.0_362)
[info] loading global plugins from /Users/mdedetrich/.sbt/1.0/plugins
[info] loading settings for project incubator-pekko-http-build from plugins.sbt ...
[info] loading project definition from /Users/mdedetrich/github/incubator-pekko-http/project
[info] compiling 1 Scala source to /Users/mdedetrich/github/incubator-pekko-http/project/target/scala-2.12/sbt-1.0/classes ...
[info] loading settings for project pekko-http from build.sbt ...
[info] resolving key references (32390 settings) ...
[info] Building Pekko HTTP 0.0.0+4290-4c0b8efe+20230221-1321-SNAPSHOT against Pekko 0.0.0+26592-864ee821-SNAPSHOT on Scala 2.13.8
[info] set current project to pekko-http-root (in build file:/Users/mdedetrich/github/incubator-pekko-http/)
[info] Setting Scala version to 2.12.15 on 21 projects.
[info] Reapplying settings...
[info] Building Pekko HTTP 0.0.0+4290-4c0b8efe+20230221-1321-SNAPSHOT against Pekko 0.0.0+26592-864ee821-SNAPSHOT on Scala 2.12.15
[info] set current project to pekko-http-root (in build file:/Users/mdedetrich/github/incubator-pekko-http/)
[error] There are files without headers!
[error]   /Users/mdedetrich/github/incubator-pekko-http/http-tests/src/multi-jvm/scala/org/apache/pekko/http/PekkoHttpServerLatencyMultiNodeSpec.scala
[error] (http-tests / Multi-jvm / headerCheck) There are files without headers!
[error]   /Users/mdedetrich/github/incubator-pekko-http/http-tests/src/multi-jvm/scala/org/apache/pekko/http/PekkoHttpServerLatencyMultiNodeSpec.scala
[error] Total time: 0 s, completed Feb 21, 2023 1:21:56 PM
```

---

I've found that `headerCreate` some times need a `clean`, that it seems to ignore files that have not been worked on recently.

---

I can see how this issue can occur locally but is this relevant for CI on a fresh checkout?

---

I am going to go ahead and merge this, there is a strong chance the discrepancy is like due to what @pjfanning stated in which case I don't see how it would be an issue for CI (can always improve this if its not the case).

---

Great, even better if it works like that.